crates/iota-core/src/authority.rs

```diff
@@ -1816,15 +1816,22 @@ impl AuthorityState {
         Ok((
             DryRunTransactionBlockResponse {
                 // to avoid cloning `transaction`, fields are populated in this order
-                suggested_gas_price: self
+                suggested_gas_price_with_ogd: self
```

Comment:
`get_prediction_suggested_gas_price()` returns the prediction for Sui's algorithm. Instead of replacing the `suggested_gas_price` field, you should keep it as it is and add an additional one referencing to `get_suggested_gas_price_with_ogd`